lib/errors/index.ts

```diff
@@ -139,12 +149,46 @@ export class ArsenalError extends Error {
             const error = value[1];
             const { code, description } = error;
             const get = () => new ArsenalError(name, code, description);
-            Object.defineProperty(errors, name, { get });
+            // Perf improvement for the ok error.
+            // As it's used mainly for HTTP response using the code and message only.
+            // Return the same instance as no stack trace or other property are needed
+            if (name === 'ok') {
+                Object.defineProperty(errors, name, { value: get() });
+            } else {
+                // Ideally for other errors caching the instance outside request handling
+                // would be beneficial if .customizeDescription is used or if
+                // no stack trace or additional field is required
+                Object.defineProperty(errors, name, { get });
+            }
         });
         return errors as Errors
     }
+
+        /**
+         * Like errors but generate singleton instances of errors
+         * To use before .customizeDescription to avoid intermediate instance.
+         * Adds +0.2 MB in heap at start
+         */
+        static errorInstances() {
+            const errors = {}
+            Object.entries(rawErrors).forEach((value) => {
+                const name = value[0] as Name;
+                const error = value[1];
+                const { code, description } = error;
+                const instance = new ArsenalError(name, code, description);
+                Object.defineProperty(errors, name, { value: instance });
+            });
+            return errors as Errors
+        }
 }
 
 /** Mapping of all possible Errors.
  * Use them with errors[error].customizeDescription for any customization. */
 export default ArsenalError.errors();
+
+/**
+ * Same as errors but already instanciated.
+ * To be used like errorInstances[error].customizeDescription or errorInstances.ok.code.
+ * Avoid intermediate instance
+ */
+export const errorInstances = ArsenalError.errorInstances();
```

Comment:
I'll try to use this where it's possible.

But if you think it's too much changes, I can drop this commit

Reply:
Do you think there is a way to keep the same variable name in the projects, to not require any code change? I believe no, but that would be the best way to be backward compatible. From what I understand this is mandatory unless you drop the commit, right?

Reply:
We can't use the same variable, we need to keep the `errors` helper for places where we want a real new error instance with stacktrace and possibility to add fields on it.

If I drop this commit, we would have no code change to do but keep a less performant intermediate instanciation when doing `errors.xx.customizeDescription`

Reply:
Actually, I just noticed branches development/8 copy the stack from the error: https://github.com/scality/Arsenal/blob/56e3db709b1202908d95f6ce1c0e5f7148064501/lib/errors/index.ts#L116

So I can't bring this pre instanciated errors into branches 8...